src/spack_commands.py

```diff
@@ -27,13 +28,28 @@ def log_with_spack(command: str,
 
     # Setup spack env
     spack_env = f'. {spack_c2sm_path}/setup-env.sh'
+    if uenv:
+        spack_env += ' /user-environment'
+
+    # Only use srun as Jenkins user and not on balfrin
+    use_srun = allow_srun and getpass.getuser() == 'jenkins' and machine_name(
+    ) != 'balfrin'
+
+    uenv_args = ''
+    if uenv:
+        uenv_mount_point = f'{uenv}:/user-environment'
+        if use_srun:
+            uenv_args = '--uenv=' + uenv_mount_point
+        else:
+            uenv_args = 'squashfs-mount ' + uenv_mount_point + ' -- '
 
     # Distribute work with 'srun'
-    if srun and getpass.getuser() == 'jenkins':
+    if use_srun:
         # The '-c' argument should be in sync with
         # sysconfig/<machine>/config.yaml config:build_jobs for max efficiency
+
+        # No entry for balfrin, trigger and error instead if requested to run with srun
```

Comment:
Balfrin used to take the same code path as any other machine. Now it is a special case and it will see errors. IMHO this is a degradation of the code. Consider reverting this.

Reply:
Because of the different ways to mount the uenv through `--uenv` arg or `squashfs-mount` I need a clear distinction between an batched and non batched test.
There is no way around that.

And for better debugabillity I think it makes sense to raise an error if for Balfrin the batched test code-path is taken.

Reply:
This code is not future prove. If it is executed on any other machine besides Tsa, Datin, Balfrin; it will fail. I think it's a better idea to make it run on any machine, and specialize Tsa and Daint to use "sruns".
And since we can't predict the srun commands of future machines, the code should fall back to use no srun in the default case.

Reply:
Ok I can revert it that way.

Reply:
Now all machines except Tsa and Daint won't use srun. Is it fine for you now?

Reply:
Yes. Thank you.